lib/client/terminal/terminal_unix.go

```
@@ -0,0 +1,160 @@
+	// Put the terminal into raw mode.
+	ts, err := term.SetRawTerminal(0)
+	if err != nil {
+		log.Warnf("Could not put terminal into raw mode: %v", err)
+	} else {
+		// Ensure the terminal is reset on exit.
+		t.closeWait.Add(1)
+		go func() {
+			<-t.closer.C
+			term.RestoreTerminal(0, ts)
+			t.closeWait.Done()
+		}()
+	}
+
+	// Convert Unix-specific signals to our abstracted events.
+	ctrlZSignal := make(chan os.Signal, 1)
+	signal.Notify(ctrlZSignal, syscall.SIGTSTP)
+	resizeSignal := make(chan os.Signal, 1)
+	signal.Notify(resizeSignal, syscall.SIGWINCH)
+	go func() {
+		for {
+			select {
+			case <-ctrlZSignal:
+				t.writeEvent(StopEvent{})
+			case <-resizeSignal:
+				t.writeEvent(ResizeEvent{})
+			}
+		}
+	}()
+
+	// NOTE: Unix does not require any special input handling.
+
+	return nil
+}
+
+// Size fetches the current terminal size as measured in columns and rows.
+func (t *Terminal) Size() (width int16, height int16, err error) {
+	size, err := term.GetWinsize(0)
+	if err != nil {
+		return 0, 0, trace.Errorf("Unable to get window size: %v", err)
+	}
+
+	return int16(size.Width), int16(size.Height), nil
+}
+
+// IsAttached determines if this terminal is attached to an interactive console
+// session.
+func (t *Terminal) IsAttached() bool {
+	return t.Stdin() == os.Stdin && term.IsTerminal(os.Stdin.Fd())
+}
+
+// Resize makes a best-effort attempt to resize the terminal window. Support
+// varies between platforms and terminal emulators.
+func (t *Terminal) Resize(width, height int16) error {
+	os.Stdout.Write([]byte(fmt.Sprintf("\x1b[8;%d;%dt", height, width)))
+
+	return nil
+}
+
+func (t *Terminal) Stdin() io.Reader {
+	return t.stdin
+}
+
+func (t *Terminal) Stdout() io.Writer {
+	return t.stdout
+}
+
+func (t *Terminal) Stderr() io.Writer {
+	return t.stderr
+}
+
+// Close closes the Terminal, restoring the console to its original state.
+func (t *Terminal) Close() error {
+	_ = t.closer.Close()
```

Why discard the error?

---

It could never return an error previously, but I've now just bubbled it up.